### PR TITLE
Create batch operation for Datastore

### PR DESCRIPTION
**Pull request by syhsiao**

### CODE REVIEW

File: store4s/src/main/scala/store4s/Datastore.scala [hunk 1]
```diff
@@ -23,21 +23,26 @@ case class Datastore(underlying: GDatastore) {
   def get(key: Key) = Option(
     underlying.get(key, Seq.empty[ReadOption]: _*)
   )
-  def get(key: Seq[Key]) = Option(
-    underlying.get(key.asJava, Seq.empty[ReadOption]: _*)
-  )
+  def get(key: Seq[Key]) = underlying.get(key.asJava, Seq.empty[ReadOption]: _*)
```
Comment (pishen) [inline store4s/src/main/scala/store4s/Datastore.scala]: Add `.asScala` to change the return type from Java Iterator to Scala Iterator.

File: store4s/src/main/scala/store4s/Datastore.scala [hunk 2]
```diff
@@ -23,21 +23,26 @@ case class Datastore(underlying: GDatastore) {
...
   def getEither[A: WeakTypeTag: EntityDecoder](name: String) = {
     get(keyFactory[A].newKey(name)).map(decodeEntity[A])
   }
   def getEither[A: WeakTypeTag: EntityDecoder](id: Long) = {
     get(keyFactory[A].newKey(id)).map(decodeEntity[A])
   }
+  def getEithers[A: WeakTypeTag: EntityDecoder](nameList: Seq[String]) = {
```
Comment (pishen) [inline store4s/src/main/scala/store4s/Datastore.scala]: To support both name and id, let's change it to `getEithersByName` and `getEithersById` then.

File: store4s/src/main/scala/store4s/Datastore.scala [hunk 3]
```diff
@@ -16,22 +17,32 @@ case class Datastore(underlying: GDatastore) {
     .setKind(weakTypeOf[A].typeSymbol.name.toString())
 
   def add(entity: FullEntity[_]) = underlying.add(entity)
+  def add(entity: Seq[FullEntity[_]]) = underlying.add(entity: _*)
```
Comment (pishen) [inline store4s/src/main/scala/store4s/Datastore.scala]: Change `entity` to `entities`?

File: store4s/src/main/scala/store4s/Datastore.scala [hunk 4]
```diff
@@ -16,22 +17,32 @@ case class Datastore(underlying: GDatastore) {
...
   def getRight[A: WeakTypeTag: EntityDecoder](name: String) = {
     getEither[A](name).map(_.toTry.get)
   }
   def getRight[A: WeakTypeTag: EntityDecoder](id: Long) = {
     getEither[A](id).map(_.toTry.get)
   }
+  def getRights[A: WeakTypeTag: EntityDecoder](name: Seq[String]) = {
+    getEithers[A](name).map(_.toTry.get)
+  }
   def delete(key: Key) = underlying.delete(key)
```
Comment (pishen) [inline store4s/src/main/scala/store4s/Datastore.scala]: Change this to `delete(key: Key*)` for batch as well?

File: store4s/src/main/scala/store4s/Datastore.scala [hunk 5]
```diff
@@ -16,22 +17,32 @@ case class Datastore(underlying: GDatastore) {
...
   def update(entity: Entity) = underlying.update(entity)
```
Comment (pishen) [inline store4s/src/main/scala/store4s/Datastore.scala]: Change this to `update(entity: Entity*)` for batch as well?

**Review by pishen — CHANGES_REQUESTED** [review]

File: store4s/src/main/scala/store4s/Datastore.scala (hunk 3 above)
Comment (syhsiao) [reply store4s/src/main/scala/store4s/Datastore.scala]: Change on put() & get() too

**Review by pishen — APPROVED** [review]

**Pull request merged**